### PR TITLE
Add an example of random number generation

### LINKED ISSUE #231: Add an example of generating random numbers
As pointed out here https://github.com/kokkos/kokkos/issues/5714#issuecomment-1360428970
there is no example to show how to generate random numbers in the wiki.

### PR DESCRIPTION
**Pull request by dalg24**
Fix #231

### CODE REVIEW

File: docs/source/API/algorithms/Random-Number.md [hunk 1]
```diff
@@ -131,3 +131,37 @@ For the selected 32-bit unsigned integer type, three range options are shown: [0
...
+
+
+## Example
+
+```c++
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+
+int main() {
+  Kokkos::Random_XorShift64_Pool<> random_pool(/*seed=*/12345);
```
Comment (crtrott) [inline docs/source/API/algorithms/Random-Number.md:143]: Add a Kokkos ScopeGuard or call initialize/finalize

File: docs/source/API/algorithms/Random-Number.md [hunk 2]
```diff
@@ -131,3 +131,37 @@ For the selected 32-bit unsigned integer type, three range options are shown: [0
...
+
+  int total = 1000000;
+  int count;
+  Kokkos::parallel_reduce(
+      "approximate_pi" total,
```
Comment (crtrott) [inline docs/source/API/algorithms/Random-Number.md:148]: , missing

**Review by crtrott — CHANGES_REQUESTED** [review]

File: docs/source/API/algorithms/Random-Number.md (hunk 1 above)
Comment (crtrott) [reply docs/source/API/algorithms/Random-Number.md:143]: You probably need to add argc/argv args to your main() to call the ScopeGuard.

**Comment by crtrott** [comment]
Other than the mentioned things looks good.

File: docs/source/API/algorithms/Random-Number.md (hunk 1 above)
Comment (dalg24) [reply docs/source/API/algorithms/Random-Number.md:143]: haha ups

**Comment by dalg24** [comment]
Fixed, thanks

**Review by crtrott — APPROVED** [review]

**Pull request merged**